### PR TITLE
Dem minor changes

### PR DESCRIPTION
**Pull request by shahabgol**
- Bugs are fixed in contact force calculations
- Normal and Tangential contact forces are validated
- Dashpot force is eliminated from particle-wall tangential force calculations
- A bug is fixed in particle-line contacts
- Two unit tests are added

### CODE REVIEW

File: applications_tests/dem_2d/circle_with_floating_walls.mpirun=1.output [hunk 1]
```diff
@@ -4,61 +4,61 @@ Starting simulation with Lethe/DEM on 1 processors
 ***************************************************************** 
 
 ***************************************************************** 
-50 particles were inserted from total 50 particles
+50 particles were inserted, 0 particles remained
```
Comment (blaisb) [inline applications_tests/dem_2d/circle_with_floating_walls.mpirun=1.output:7]: 0 particles remained -> should be 0 particles **remains**

File: applications_tests/dem_2d/circle_with_floating_walls.mpirun=2.output [hunk 2]
```diff
@@ -4,61 +4,61 @@ Starting simulation with Lethe/DEM on 2 processors
 ***************************************************************** 
 
 ***************************************************************** 
-50 particles were inserted from total 50 particles
+50 particles were inserted, 0 particles remained
```
Comment (blaisb) [inline applications_tests/dem_2d/circle_with_floating_walls.mpirun=2.output:7]: Same as above, clearly this is in the code. I won't comment on that anymore :)

File: source/dem/insertion.cc [hunk 3]
```diff
@@ -31,9 +31,8 @@ Insertion<dim>::print_insertion_info(const unsigned int inserted_this_step,
       std::cout
         << "***************************************************************** "
            "\n";
-      std::cout << inserted_this_step << " particles were inserted from total "
-                << inserted_this_step + remained_particles << " particles"
-                << std::endl;
+      std::cout << inserted_this_step << " particles were inserted, "
+                << remained_particles << " particles remained" << std::endl;
```
Comment (blaisb) [inline source/dem/insertion.cc:35]: Modify text of this line. Maybe particles remaining instead is better.

File: source/dem/particle_point_line_contact_force.cc [hunk 4]
```diff
@@ -40,8 +40,9 @@ ParticlePointLineForce<dim>::calculate_particle_point_line_contact_force(
       // are used to consider non-linear relation of the contact force to
       // the normal overlap
       double model_parameter_betha =
-        log(physical_properties.poisson_ratio_wall) /
-        sqrt(pow(log(physical_properties.poisson_ratio_wall), 2.0) + 9.8696);
+        log(physical_properties.restitution_coefficient_wall) /
+        sqrt(pow(log(physical_properties.restitution_coefficient_wall), 2.0) +
```
Comment (blaisb) [inline source/dem/particle_point_line_contact_force.cc:44]:
```suggestion
        sqrt(pow(log(physical_properties.restitution_coefficient_wall), 2) +
```

When you have a power 2, use 2. The compiler can then optimize it more.

File: source/dem/pp_nonlinear_force.cc [hunk 5]
```diff
@@ -262,44 +264,48 @@ PPNonLinearForce<dim>::calculate_nonlinear_contact_force_and_torque(
   double normal_spring_constant =
     1.3333 * effective_youngs_modulus * radius_times_overlap_sqrt;
   double normal_damping_constant =
-    -1.8257 * model_parameter_beta * sqrt(model_parameter_sn * effective_mass);
+    -1.8257 * model_parameter_betha * sqrt(model_parameter_sn * effective_mass);
```
Comment (blaisb) [inline source/dem/pp_nonlinear_force.cc:267]:
```suggestion
    -1.8257 * model_parameter_beta * sqrt(model_parameter_sn * effective_mass);
```
Beta does not take an h

File: source/dem/pp_nonlinear_force.cc [hunk 6]
```diff
@@ -245,11 +245,13 @@ PPNonLinearForce<dim>::calculate_nonlinear_contact_force_and_torque(
...
-    Poisson_ratio_particle_log /
-    sqrt(Poisson_ratio_particle_log * Poisson_ratio_particle_log + 9.8696);
+  const double restitution_coefficient_particle_log =
+    std::log(physical_properties.restitution_coefficient_particle);
+  const double model_parameter_betha =
+    restitution_coefficient_particle_log /
+    sqrt(restitution_coefficient_particle_log *
+           restitution_coefficient_particle_log +
+         9.8696);
   const double radius_times_overlap_sqrt =
```
Comment (blaisb) [inline source/dem/pp_nonlinear_force.cc:255]: Good idea to pre-calculate those.

File: source/dem/pp_nonlinear_force.cc [hunk 7]
```diff
@@ -245,11 +245,13 @@ PPNonLinearForce<dim>::calculate_nonlinear_contact_force_and_torque(
   // Calculation of model parameters (betha, sn and st). These values
```
Comment (blaisb) [inline source/dem/pp_nonlinear_force.cc:245]:
```suggestion
  // Calculation of model parameters (beta, sn and st). These values
```

File: source/dem/pp_nonlinear_force.cc [hunk 8]
```diff
@@ -262,44 +264,48 @@ PPNonLinearForce<dim>::calculate_nonlinear_contact_force_and_torque(
   double normal_spring_constant =
     1.3333 * effective_youngs_modulus * radius_times_overlap_sqrt;
   double normal_damping_constant =
-    -1.8257 * model_parameter_beta * sqrt(model_parameter_sn * effective_mass);
+    -1.8257 * model_parameter_betha * sqrt(model_parameter_sn * effective_mass);
   double tangential_spring_constant =
     8.0 * effective_shear_modulus * radius_times_overlap_sqrt + DBL_MIN;
   double tangential_damping_constant =
-    -1.8257 * model_parameter_beta * sqrt(model_parameter_st * effective_mass);
+    -1.8257 * model_parameter_betha * sqrt(model_parameter_st * effective_mass);
```
Comment (blaisb) [inline source/dem/pp_nonlinear_force.cc:271]:
```suggestion
    -1.8257 * model_parameter_beta * sqrt(model_parameter_st * effective_mass);
```

File: tests/dem/post_collision_velocity.output [hunk 9]
```diff
@@ -0,0 +1,2 @@
+
+DEAL::Coefficient of restitution is 0.7 and the velocity of particle before collision is 0.1, the velocity of particle after collision is: 0.0662483
```
Comment (blaisb) [inline tests/dem/post_collision_velocity.output:2]: Shouldnt it be 0.07 and not 0.066? Why is there such a difference? Does it go down when you decrease the time step?

**Review by blaisb — COMMENTED** [review]
Very minor comments to look at. Otherwise, good job.

File: include/dem/find_boundary_cells_information.h [hunk 10]
```diff
@@ -55,10 +55,10 @@ class BoundaryCellsInformation
...
 
   /**
-   * @brief The build fonction builds all the boundary cell information structure
-   * that find which cell has a boundary face, boundary line or a boundary
-   * point. Additionnaly, it identifies which cell are located in the vicinity
-   * (controlled by the max cell diameter of flaoting  walls
+   * @brief The build fonction builds all the boundary cell information
+   * structure that find which cell has a boundary face, boundary line or a
+   * boundary point. Additionnaly, it identifies which cell are located in the
+   * vicinity (controlled by the max cell diameter of flaoting  walls
```
Comment (tonielgeitani) [inline include/dem/find_boundary_cells_information.h:61]: of floating

File: source/dem/pw_contact_force.cc [hunk 11]
```diff
@@ -89,19 +89,8 @@ PWContactForce<dim>::update_contact_information(
   // which were already in contact (pairs_in_contact) needs to
   // modified using its history, while the tangential_overlaps of
```
Comment (tonielgeitani) [inline source/dem/pw_contact_force.cc:90]: be modified

**Pull request merged**